pageserver/src/layered_repository/interval_tree.rs

```diff
@@ -230,6 +238,35 @@ where
     }
 }
 
+impl<'a, I> DoubleEndedIterator for IntervalIter<'a, I>
+where
+    I: IntervalItem + ?Sized,
+{
+    fn next_back(&mut self) -> Option<Self::Item> {
+        // Iterate over all elements in all the points in 'point_iter'. To avoid
+        // returning the same element twice, we only return each element at its
+        // starting point.
+        loop {
+            // Return next remaining element from the current point
+            if let Some((point_key, elem_iter)) = &mut self.elem_iter {
```

Comment:
Logically we need to traverse overlapped interval in reverse order too.
But for counting layers it is not important, because we are looking at start LSN in any case and all elements returned by elem_iterator has the same start LSN.